File: tutorials/testing/c++/us_map_test.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/USState.h"
#include "data_src/USCounty.h"
#include "USMap.h"

using namespace bridges;

// this program illustrates how to access the data of the US state county
// maps and apply attributes to them
int main(int argc, char **argv) {

	// create bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

	// set title
	bridges.setTitle("Accessing US State County Maps");

	DataSource ds(&bridges);

	cout << "Retrieving US State County Map Data" << endl;


	vector<string> states = {"Vermont", "Iowa"};
	vector<bridges::dataset::USState> 
		map_data = ds.getUSStateCountyMapData (states);


	// 
	// now do what needs to be done as part of your assignment
	// 

	// this will prepare the dataset for visualization
	// print info
	for (auto& st : map_data) {
		cout << "State: " << st.getStateName() << "\n";
		unordered_map<string, County> counties = st.getCounties();
		st.setViewCountiesFlag(false);
		int k = 0; 
		for (auto c : counties) {
			c.second.setHideFlag(true);
			if (k++ < 10)
			cout <<  "\t" << c.second.getCountyName() << endl; 
		}
	}
	USMap us_maps(map_data);

	bridges.setDataStructure(&us_maps);
	bridges.visualize();

	return 0;
}

File: tutorials/testing/java/us_map_test.java
import java.util.Vector;
import java.util.Map;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.State;
import bridges.data_src_dependent.County;
import bridges.base.SLelement;
import bridges.base.USMap;
import bridges.base.Color;


// This program fragment illustrates how to access the US Cities  data
public class us_map_test {
	public static void main(String[] args) throws Exception {

		// create Bridges object
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
		// read the earth quake  data and build the BST
		bridges.setTitle("Accessing US State County Maps)");

		DataSource ds = bridges.getDataSource();

		// case 1 : map and counties  of  a few states
		bridges.setDescription("3 US states, with and without counties");
		String[] states = {"Vermont", "Iowa", "North Carolina"};
	//	String[] states = {"North Carolina"};
		Vector<State> map_data = ds.getUSMapCountyData (states, true);

		State s = map_data.get(0);
		s.setViewCountiesFlag(true);
		s.setStrokeColor(new Color(255,0,0));
		s.setFillColor(new Color(0,255,0));
		s.setViewCountiesFlag(false);

		{
			s = map_data.get(1);

			for (Map.Entry<String,County> e: s.accessCounties().entrySet()) {
				County c = e.getValue();
				c.setStrokeColor(new Color(50,250,50));
				c.setFillColor(new Color(0,0,25));
			}
		}

		USMap us_maps = new USMap(map_data);
		bridges.setMap(us_maps);
    
		SLelement<String>  el0 = new SLelement<String>("Charlotte", "Go Niners!");
		el0.setLocation(-80.8431, 35.2271); //35.2271N, 80.8431W
		bridges.setDataStructure(el0);
		bridges.visualize();

		// case 2:  all states, no counties
		bridges.setDescription("All US states, No Counties");
		map_data = ds.getUSMapData(); // all states, no counties
		us_maps.setStateData(map_data);
		bridges.setMap(us_maps);
		bridges.visualize();

		// case 3:  all states, all counties
		bridges.setDescription("All US states, All Counties");
		map_data = ds.getUSMapCountyData (); // all states, no counties
		us_maps.setStateData(map_data);
		bridges.setMap (us_maps);
		bridges.visualize();
	}
}
